### ISSUE TITLE
An Id shouldn't be required when calling 'create'

### ISSUE THREAD

**Issue by ghost** [jpillora/jquery.rest#13]
Currently when using the restClient to create, if an Id isn't passed as the first argument, an error is thrown stating 
'Uncaught ERROR: jquery.rest: Invalid argument: [object Object](object). Must be strings or ints (IDs) followed by one optional plain object (data). '

When creating a new instance of a resource, which I imagine should be a POST, I don't yet know what the Id of the object will be.

**Comment by ghost** [comment]
I realized that the reason this wasn't working was due to a check to make sure the object is a 'plainObject', I replaced this $.isPlainObject(arg) with (args instanceof Object) and it seems to have resolved the issue. 

Is there a reason that you would advise against this, or should that work just fine?

**Comment by jpillora** [comment]
Hmmm, I think there is a misunderstanding somewhere, maybe it would help if you posted the code you hope would work?

> When creating a new instance of a resource, which I imagine should be a POST, I don't yet know what the Id of the object will be.

I assume you mean when you send a **C**reate POST request (not creating a new foo resource instance - `client.add('foo')`). It's true that you don't know the ID of an item not yet created, thats why `client.foo.create({height:180})` doesn't require an ID, though `client.foo.update('bar',{height:180})` does

Don't think this is a bug so I'll close, may reopen with more information.

**Comment by ghost** [comment]
```
define(
    ['models/customTypeModel'],
    function (CustomTypeModel) {
        var makeCustomTypeModel = function (customTypeData) {
            if (customTypeData instanceof Array) {
                return customTypeData.map(makeCustomTypeModel);
            }
            return new CustomTypeModel(customTypeData);
        }

        var CustomTypeService = function () {
            var root = this;
            var restClient = new $.RestClient('/api/');
            restClient.add('customTypes');

            this.create = function (customTypeModel) {
                return restClient
                    .customTypes
                    .create(customTypeModel)
                    .then(makeCustomTypeModel);
            };
        };
        return new CustomTypeService();
    });
```

This code will fail because I am passing an object that is not a 'plainObject'(i.e. on created with an object literal or `new Object()`), I am passing an object created with `new CustomTypeModel()`.

When the RestClient library gets to the point where it checks for an id, it checks to see if the data getting passed in is a plainObject, which it isn't. However, removing this explicit check for a plain object and simply checking against `instanceof Object` resolves this issue.
`282:        } else if ($.isPlainObject(arg) && data === null) {`
`282:        } else if ((arg instanceof Object) && data === null) {`

Here are two errors I encountered. The first is from passing in a customTypeModel and the second is from passing in an array.
`Uncaught ERROR: jquery.rest: Invalid argument: [object Object] (object). Must be strings or ints (IDs) followed by one optional plain object (data).`

`Uncaught ERROR: jquery.rest: Invalid argument:  (array). Must be strings or ints (IDs) followed by one optional plain object (data).`

**Comment by jpillora** [comment]
Ahhh I see, it's a model object which isn't plain, okay, I'll dumb it down to `typeof "object"`, which should fix it

**Comment by ghost** [comment]
:) Thanks
